### DISCUSSION TITLE
How to make vcpkg in manifest mode only update when cmake find_* function is called?

### DISCUSSION THREAD

**Discussion by MarkCallow** [microsoft/vcpkg#39928] [Q&A] [answered]
In my project I have two optional components that require some dependencies that the rest of the project components do not. I'm setting up vcpkg manifest mode to get those dependencies. Regardless of whether or not those components are selected vcpkg updates itself and the installed packages at the start of every config. I had hoped the relevant cmake `find_*` commands would trigger vcpkg to do its stuff. I do not want people who are not building those optional components to suffer the delay caused by the update. Is there a way to limit vcpkg to just when the optional components are selected?

I have found `VCPKG_MANIFEST_INSTALL` and have set that to false which does prevent the update but I am worried it prevents all updates even when the components are selected and the relevant `find_*` commands are called.

I suppose I could set VCPKG_MANIFEST_INSTALL` true when one of the components is selected. Is that a reasonable approach?

I originally tried to put `vcpkg.json` in the project subdirectory containing the CMakeLists.txt for the optional components but vcpkg didn't find it so now I have it in the project root. I subsequently learned about `VCPKG_MANIFEST_DIR`. Now I am wondering if there is any benefit moving it back to the subdirectory and setting that variable.

**Comment by AugP** [comment]
You can configure some features for your project to enable this. 

See: https://learn.microsoft.com/vcpkg/concepts/features#example-3-multiple-related-projects-in-a-single-repository

Assuming your optional components have CMake options associated with them, you will need to write some logic to enable the required vcpkg manifest mode features when an option is set. This needs to happen before 'project()' is called.

**Reply by MarkCallow** [reply]
Thank you for your help.

**Comment by autoantwort** [comment, marked as answer]
Use https://learn.microsoft.com/en-us/vcpkg/users/buildsystems/cmake-integration#vcpkg_manifest_features

**Reply by MarkCallow** [reply]
Thank you for your help. I chose this as the answer. The link provided by @AugP left me with some questions while this link answered those and everything else in its short self-contained example.
